apps/backend/src/tests/response_api.test.ts

```diff
@@ -0,0 +1,555 @@
+      .send(updateData);
+
+    expect(httpResponse.status).toBe(200);
+    expect(httpResponse.body.id).toBe(ANTONINAS_RESPONSE_ID);
+    expect(httpResponse.body.status).toBe('PENDING');
+    expect(httpResponse.body.content).toBe(updateData.content);
+
+    const finalResponseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(finalResponseInDb.id).toBe(initialResponseInDb.id);
+    expect(finalResponseInDb.request_id).toBe(initialResponseInDb.request_id);
+    expect(finalResponseInDb.user_id).toBe(initialResponseInDb.user_id);
+    expect(finalResponseInDb.status).toBe(initialResponseInDb.status);
+    expect(finalResponseInDb.content).toBe(updateData.content);
+  });
+});
+
+describe('Tests for deleting a request: DELETE /responses/:id', () => {
+  let token: string;
+
+  beforeAll(async () => {
+    const loginResponse: Response = await loginUser(ANTONINAS_LOGIN_DATA);
+    token = loginResponse.body.token;
+  });
+
+  beforeEach(async () => {
+    await seed();
+  });
+
+  test("User cannot delete request if they aren't logged in", async () => {
+    const httpResponse: Response = await api.delete(
+      `/api/responses/${ANTONINAS_RESPONSE_ID}`,
+    );
+
+    const responseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.status).toEqual(401);
+    expect(httpResponse.body.error).toBe('user not signed in');
+    expect(responseInDb).not.toBe(null);
+  });
+
+  test('Delete non-existent request fails', async () => {
+    const NON_EXISTENT_ID = 1000;
+
+    const httpResponse: Response = await api
+      .delete(`/api/responses/${NON_EXISTENT_ID}`)
+      .set('Authorization', `Bearer ${token}`);
+
+    expect(httpResponse.status).toBe(404);
+    expect(httpResponse.body.error).toBe('No Response found');
+  });
+
+  test('User cannot delete request if not its creator or admin', async () => {
+    const loginResponse = await loginUser(BOBS_LOGIN_DATA);
+    const bobsToken: string = loginResponse.body.token;
+
+    const httpResponse: Response = await api
+      .delete(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${bobsToken}`);
+
+    const responseInDb = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.status).toEqual(401);
+    expect(responseInDb).not.toBe(null);
+  });
+
+  test('Delete an existing response as the creator', async () => {
+    const httpResponse: Response = await api
+      .delete(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`);
+
+    let deleted;
+    try {
+      deleted = await testHelpers.getSingleResponse(ANTONINAS_RESPONSE_ID);
+    } catch (error) {
+      deleted = null;
+    }
+
+    expect(httpResponse.status).toEqual(204);
+    expect(deleted).toBe(null);
+  });
+
+  // In the Test Data, The Creator of the Response and Admin of neighborhood
```

Comment:
I think we should add more test data (or modify the current) to test for that

Reply:
done.